File: src/main/java/com/market/ddanggyeo/user/dto/UserDto.java
package com.market.ddanggyeo.user.dto;

import lombok.RequiredArgsConstructor;

import javax.persistence.*;

@Entity
@RequiredArgsConstructor
public class UserDto {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;
    private String email;
    private String username;
    private String phone;
    @Embedded
    private Address address;
}

File: src/main/kotlin/com/market/ddanggyeo/user/dto/UserDto.kt
package com.market.ddanggyeo.user.dto

import javax.persistence.*

@Entity
class UserDto(email: String, username: String, phone: String, address: Address) {

    @Id
    var email: String = email

    var username: String? = username

    var phone: String? = phone

    @get:Embedded
    var address: Address? = address
}